### ISSUE TITLE
Segmentation fault on a small SV example

### ISSUE THREAD

**Issue by jmonlong** [DecodeGenetics/graphtyper#28]
I'm getting `Segmentation fault (core dumped)` when I try to genotype SVs on a small simulated dataset. 
Do you see anything that could cause this from my input files?

[simsv.zip](https://github.com/DecodeGenetics/graphtyper/files/4316292/simsv.zip) contains:

- `ref.fa`
- `truth.vcf.gz`
- `s0.bam`

I tried running the following commands:

```
graphtyper genotype_sv ref.fa truth.vcf.gz --sam=s0.bam --region=chr1:1-2852430
```

and 

```
graphtyper construct my_graph ref.fa chr1 --vcf=truth.vcf.gz --sv_graph
graphtyper index my_graph
mkdir calls
graphtyper call my_graph . --sam=s0.bam --no_new_variants --output=calls
```

I'm using the v2.1 graphtyper binary.

**Comment by hannespetur** [comment]
Thank you for posting your problem and giving me data to reproduce it. It seems the problem is that graphtyper is mistakenly not ignoring the secondary/duplicated/supplementary reads in the BAM file in the "genotype_sv" command. All those reads are not present in the data I work with so the bug got through my internal tests. I will try to push a fix for this later today.

Best,
Hannes

**Comment by hannespetur** [comment]
Hmm well, there seems to be something more going on than that. I will try to try to find a fix asap, but probably not happening today.

Best,
Hannes

**Comment by jmonlong** [comment]
No worries, thanks for looking into it!

**Comment by hannespetur** [comment]
Hey, I think I've found a proper fix released in version 2.2. Though, GraphTyper is not properly parsing the inversions in your VCF, could you reformat them with a SVTYPE=INV;SVSIZE=<size> in the INFO field, like this:

```sh
zcat truth.vcf.gz |
    awk -v OFS=$'\t' '
NR==2{print "##INFO=<ID=SVTYPE,Number=1,Type=String,Description=\"SV type\">"; print "##INFO=<ID=SVSIZE,Number=1,Type=Integer,Description=\"SV size\">"}
$3 !~ "inv"{print}
$3 ~ "inv"{$8=$8";SVTYPE=INV;SVSIZE="length($4)-1; $4=substr($4,1,1); $5="<INV>"; print $0}
' | bgzip -c > truth.new.vcf.gz
tabix truth.new.vcf.gz
```

Can you try version 2.2?


Thanks,
Hannes

**Comment by jmonlong** [comment]
Thanks for the fast fix, it looks like it's working. I'll give graphtyper-pipelines a try now, to prepare running on a larger datasets.
